### PR TITLE
update_kernel: Implement installation of specific kernel version

### PR DESCRIPTION
**Pull request by mdoucha**
Implement installation of specific kernel version given in `KERNEL_VERSION` job variable. Transactional systems are also supported, which required some changes to the transactional utils library.

- Related ticket: N/A
- Needles: N/A
- Verification run:
  - SLEM-5.1: https://openqa.suse.de/tests/15225298
  - SLE-12SP5: https://openqa.suse.de/tests/15225299
  - kubevirt tests (for `enter_trup_shell()` in VNC): https://openqa.suse.de/tests/15225336

### CODE REVIEW

**Comment by mdoucha** [comment]
This PR needs to be rebased on #19756.

File: tests/kernel/update_kernel.pm [hunk 1]
```diff
@@ -28,12 +28,14 @@ use package_utils;
 sub check_kernel_package {
     my $kernel_name = shift;
 
+    enter_trup_shell(global_options => '-c') if is_transactional;
```
Comment (czerw) [inline tests/kernel/update_kernel.pm:31]: If is system rebooted , function works even on transactional system and there is no change needed.  If i remember correctly we decided to do reboot when we introduced support for transactional systems in `update_kernel`. 

https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/7eefe7bbf87329880a2cf786271bf1b6aa174380/tests/kernel/update_kernel.pm#L114

But we can proceed with your suggestion.

**Comment by czerw** [comment]
LGTM so far, do you expect to add more changes?

File: tests/kernel/update_kernel.pm (hunk 1 above)
Comment (mdoucha) [reply tests/kernel/update_kernel.pm:31]: `check_kernel_package()` is called from multiple places, some of which reboot *after* the check.

**Comment by mdoucha** [comment]
I'm not planning any further changes but I still haven't been able to clone a successful verification run for `enter_trup_shell` changes in VNC. The kubevirt job keeps failing during machine reboot.

**Comment by mdoucha** [comment]
All verification runs finished with trup shell behaving as expected. The kubevirt job failed some tests but I can reproduce the same failure without `CASEDIR` so it appears to be a product issue unrelated to this PR.
https://openqa.suse.de/tests/15226096#step/kubevirt_tests_server/548

**Review by czerw — APPROVED** [review]

**Pull request merged**